xds/src/test/java/io/grpc/xds/XdsNamResolverTest.java

```diff
@@ -126,7 +159,23 @@ private static void assertHardCodedServiceConfig(ResolutionResult actualResult)
                 Collections.singletonMap("round_robin", Collections.EMPTY_MAP)));
   }
 
-  private XdsNameResolver newResolver(String name) {
-    return new XdsNameResolver(name);
+  @SuppressWarnings("unchecked")
+  private static void assertBootstrapServiceConfig(ResolutionResult actualResult) {
+    assertThat(actualResult.getAddresses()).isEmpty();
+    Map<String, ?> serviceConfig =
+        actualResult.getAttributes().get(GrpcAttributes.NAME_RESOLVER_SERVICE_CONFIG);
+    List<Map<String, ?>> rawLbConfigs =
+        (List<Map<String, ?>>) serviceConfig.get("loadBalancingConfig");
+    Map<String, ?> xdsLbConfig = Iterables.getOnlyElement(rawLbConfigs);
+    assertThat(xdsLbConfig.keySet()).containsExactly("xds_experimental");
+    Map<String, ?> rawConfigValues = (Map<String, ?>) xdsLbConfig.get("xds_experimental");
+    assertThat(rawConfigValues)
+        .containsExactly(
+            "balancer_name",
+            "fake_server_uri",
+            "childPolicy",
+            Collections.singletonList(
+                Collections.singletonMap("round_robin", Collections.EMPTY_MAP)));
+    assertThat(actualResult.getAttributes().get(XDS_NODE)).isSameInstanceAs(bootstrapNode);
```

Comment:
Use `isEqualTo()`.

Reply:
Why? I purposefully used `isSameInstanceAs`

Reply:
`node` is a message, it's meaningless (or more strictly it's wrong) to insist on the identity equality.

Reply:
The attribute value is intended to be a reference, for now it's Node, in the future it is XdsClient. I don't care much for Node, but we will back change to `isSameInstanceAs` for XdsClient in tests.